scrapy_poet/overrides.py

```diff
@@ -89,35 +70,64 @@ class OverridesRegistry(OverridesRegistryBase):
     def from_crawler(cls, crawler: Crawler) -> Crawler:
         return cls(crawler.settings.getlist("SCRAPY_POET_OVERRIDES", []))
 
-    def __init__(self, rules: Optional[Iterable[RuleFromUser]] = None) -> None:
-        self.rules: List[ApplyRule] = []
-        self.matcher: Dict[Type[ItemPage], URLMatcher] = defaultdict(URLMatcher)
-        for rule in rules or []:
-            self.add_rule(rule)
-        logger.debug(f"List of parsed ApplyRules:\n{self.rules}")
-
-    def add_rule(self, rule: RuleFromUser) -> None:
-        if isinstance(rule, (tuple, list)):
-            if len(rule) != 3:
-                raise ValueError(
-                    f"Invalid rule: {rule}. Rules as tuples must have "
-                    f"3 elements: (1) the pattern, (2) the PO class used as a "
-                    f"replacement and (3) the PO class to be replaced."
-                )
-            pattern, use, instead_of = rule
-            rule = ApplyRule(
-                for_patterns=Patterns([pattern]), use=use, instead_of=instead_of
-            )
-        self.rules.append(rule)
-        # FIXME: This key will change with the new rule.to_return
-        self.matcher[rule.instead_of].add_or_update(  # type: ignore
-            len(self.rules) - 1, rule.for_patterns
+    def __init__(self, rules: Optional[Iterable[ApplyRule]] = None) -> None:
+        super().__init__(rules=rules)
+        self.overrides_matcher: Dict[Type[ItemPage], URLMatcher] = defaultdict(
+            URLMatcher
         )
+        self.item_matcher: Dict[Optional[Type], URLMatcher] = defaultdict(URLMatcher)
```

Comment:
I think we should rename "item_matcher" to "item_matchers", and "overrides_matcher" to "overrides_matchers". 

I was reading code later, and for a minute thought that self.overrides_matcher is an UrlMatcher instance.

Reply:
I'll address this in a separate PR in `web-poet` which migrates all of these functionalities.

Reply:
Fixed in https://github.com/scrapinghub/web-poet/pull/112